app/src/org/commcare/dalvik/dialogs/AlertDialogFactory.java

```diff
@@ -30,6 +30,7 @@ public AlertDialogFactory(Activity context, String title, String msg) {
         messageView.setText(msg);
 
         this.dialog = builder.create();
+        dialog.setView(view);
```

Comment:
@phillipm Oops just made a comment about this on the other PR before I saw this. Yes, it needs to be here because there are methods that can be called on an AlertDialogFactory in between it's construction and calling showDialog() that change the view. Seems like the solution to the issue you're talking about is to call setView() in both getDialog() and showDialog()